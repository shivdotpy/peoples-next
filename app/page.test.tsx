import { render, screen } from '@testing-library/react';
import Home from './page';
import RootLayout from './layout';

describe('Home Page - Landing Page', () => {
  test('renders the heading', () => {
    render(<Home />);
    const heading = screen.getByText('Welcome to peoples page');
    expect(heading).toBeInTheDocument();
  });

  test('renders the link with correct text content', () => {
    render(<Home />);
    const link = screen.getByText('Explore Peoples');
    expect(link).toBeInTheDocument();
    expect(link.getAttribute('href')).toBe('/peoples');
  });

  test('renders the image with correct attributes', () => {
    render(<Home />);
    const image = screen.getByAltText('peoples');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('height', '500');
    expect(image).toHaveAttribute('width', '500');
  });

  test('renders the footer text', () => {
    render(<Home />);
    const footer = screen.getByText(
      'Â© 2023 Peoples App. No rights reserved. ðŸ˜‚'
    );
    expect(footer).toBeInTheDocument();
  });
});
