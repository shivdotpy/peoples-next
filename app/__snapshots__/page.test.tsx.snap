// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Renders home page and show text Peoples 1`] = `
<div>
  <main>
    Peoples
  </main>
</div>
`;
